src/libraries/Microsoft.Extensions.Hosting.Systemd/src/SystemdHelpers.cs

```diff
@@ -33,15 +36,32 @@ private static bool CheckParentIsSystemd()
             try
             {
                 // Check whether our direct parent is 'systemd'.
+                // This will not work for containerised applications, since our direct parent will be some container runtime
                 int parentPid = Interop.libc.GetParentPid();
                 string ppidString = parentPid.ToString(NumberFormatInfo.InvariantInfo);
                 byte[] comm = File.ReadAllBytes("/proc/" + ppidString + "/comm");
-                return comm.AsSpan().SequenceEqual(Encoding.ASCII.GetBytes("systemd\n"));
+                if (comm.AsSpan().SequenceEqual(Encoding.ASCII.GetBytes("systemd\n")))
+                {
+                    return true;
+                }
             }
             catch
             {
             }
 
+            // Convention for containers is to have the PID of the containerised process set to 1
+            if (Environment.ProcessId == 1)
+            {
+                if(!string.IsNullOrEmpty(Environment.GetEnvironmentVariable(NOTIFY_SOCKET_ENVVAR_KEY)))
```

Comment:
We can combine these and write them as:
```
```
return !string.IsNullOrEmpty(Environment.GetEnvironmentVariable(NOTIFY_SOCKET_ENVVAR_KEY) ||
          !string.IsNullOrEmpty(Environment.GetEnvironmentVariable(LISTEN_PID_ENVVAR_KEY);
```

Reply:
Actioned